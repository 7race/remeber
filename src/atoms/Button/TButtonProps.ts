export type TButtonProps = {
  children: string;
  variant: 'success' | 'submit' | 'cancel';
  outlined?: boolean;
};
