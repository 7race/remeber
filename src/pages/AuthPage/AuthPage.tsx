import { Button } from '../../atoms/Button/Button';

export const AuthPage = () => (
  <Button variant='success' outlined>
    Войти
  </Button>
);
