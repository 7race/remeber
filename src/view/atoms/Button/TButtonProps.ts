import { MouseEventHandler } from 'react';

export type TButtonProps = {
  children: string;
  variant: 'success' | 'secondary' | 'cancel';
  type?: 'submit';
  outlined?: boolean;
  onClick?: MouseEventHandler;
};
