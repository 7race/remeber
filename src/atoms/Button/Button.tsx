import Ripples from 'react-ripples';
import * as S from './Button.style';
import { TButtonProps } from './TButtonProps';

export const Button = ({ children, variant, outlined }: TButtonProps) => (
  <Ripples>
    <S.Button variant={variant} outlined={outlined}>
      {children}
    </S.Button>
  </Ripples>
);
