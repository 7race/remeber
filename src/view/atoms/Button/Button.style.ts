import styled, { css, DefaultTheme, FlattenInterpolation, ThemeProps } from 'styled-components';
import { TButtonProps } from './TButtonProps';

type TButtonType = Record<string, FlattenInterpolation<ThemeProps<DefaultTheme>>>;

const outlinedBtns: TButtonType = {
  outlineSuccess: css`
    border: 2px solid ${({ theme }) => theme.colors.green};
    color: ${({ theme }) => theme.colors.black};
    background-color: ${({ theme }) => theme.colors.white};
  `,
  outlineSubmit: css`
    border: 2px solid ${({ theme }) => theme.colors.violet};
    color: ${({ theme }) => theme.colors.black};
    background-color: ${({ theme }) => theme.colors.white};
  `,
  outlineCancel: css`
    border: 2px solid ${({ theme }) => theme.colors.red};
    color: ${({ theme }) => theme.colors.black};
    background-color: ${({ theme }) => theme.colors.white};
  `,
};

const paintedOverBtns: TButtonType = {
  success: css`
    background-color: ${({ theme }) => theme.colors.green};
  `,
  submit: css`
    background-color: ${({ theme }) => theme.colors.violet};
  `,
  cancel: css`
    background-color: ${({ theme }) => theme.colors.red};
  `,
};

const variants = {
  success: css<TButtonProps>`
    ${({ outlined }) => (outlined ? outlinedBtns.outlineSuccess : paintedOverBtns.success)};
  `,
  submit: css<TButtonProps>`
    ${({ outlined }) => (outlined ? outlinedBtns.outlineSubmit : paintedOverBtns.submit)};
  `,
  cancel: css<TButtonProps>`
    ${({ outlined }) => (outlined ? outlinedBtns.outlineCancel : paintedOverBtns.cancel)};
  `,
};

export const Button = styled.button<TButtonProps>`
  border: none;
  cursor: pointer;
  border-radius: 8px;
  font-family: 'Roboto', sans-serif;
  color: ${({ theme }) => theme.colors.white};
  width: 320px;
  height: 48px;
  ${({ variant }) => variants[variant]};
  @media ${({ theme }) => theme.media.tablet} {
    width: 384px;
    height: 48px;
  }
  @media ${({ theme }) => theme.media.notebook} {
    width: 412px;
    height: 64px;
    font-size: 24px;
  }
`;
