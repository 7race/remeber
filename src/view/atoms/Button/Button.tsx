import Ripples from 'react-ripples';
import * as S from './Button.style';
import { TButtonProps } from './TButtonProps';

export const Button = ({ children, variant, outlined, type, onClick }: TButtonProps) => (
  <Ripples>
    <S.Button variant={variant} outlined={outlined} type={type} onClick={onClick}>
      {children}
    </S.Button>
  </Ripples>
);
