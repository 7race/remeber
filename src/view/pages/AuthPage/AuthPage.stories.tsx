import { ComponentStory, ComponentMeta } from '@storybook/react';
import { AuthPage } from './AuthPage';

export default {
  title: 'AuthPage',
  component: AuthPage,
} as ComponentMeta<typeof AuthPage>;

const Template: ComponentStory<typeof AuthPage> = () => <AuthPage />;

export const Default = Template.bind({});
