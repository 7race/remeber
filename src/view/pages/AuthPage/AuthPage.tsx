import { Form } from '@molecules/Form';
import * as S from './AuthPage.styled';

export const AuthPage = () => (
  <S.AuthPage>
    <S.GlobalStyle />
    <Form />
  </S.AuthPage>
);
