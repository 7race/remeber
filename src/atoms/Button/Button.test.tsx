import { render, screen } from '../../tests-utils';
import { AuthPage } from '../../pages/AuthPage/AuthPage';
import { Button } from './Button';

describe('button in auth page', () => {
  it('render', () => {
    render(<AuthPage />);
    const btn = screen.getByRole('button', { name: 'Войти' });
    expect(btn).toBeInTheDocument();
  });

  it('correct styles: variant=success', () => {
    const { container } = render(<Button variant='success'>Войти</Button>);
    expect(container).toMatchSnapshot();
  });

  it('correct styles: variant=success outlined', () => {
    const { container } = render(
      <Button variant='success' outlined>
        Войти
      </Button>
    );
    expect(container).toMatchSnapshot();
  });

  it('correct styles: variant=submit', () => {
    const { container } = render(<Button variant='submit'>Войти</Button>);
    expect(container).toMatchSnapshot();
  });

  it('correct styles: variant=submit outlined', () => {
    const { container } = render(
      <Button variant='submit' outlined>
        Войти
      </Button>
    );
    expect(container).toMatchSnapshot();
  });

  it('correct styles: variant=cancel', () => {
    const { container } = render(<Button variant='cancel'>Войти</Button>);

    expect(container).toMatchSnapshot();
  });

  it('correct styles: variant=cancel outlined', () => {
    const { container } = render(
      <Button variant='cancel' outlined>
        Войти
      </Button>
    );
    expect(container).toMatchSnapshot();
  });
});
