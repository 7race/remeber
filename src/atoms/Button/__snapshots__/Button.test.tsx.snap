// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button in auth page correct styles: variant=cancel 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ YFRPM"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;

exports[`button in auth page correct styles: variant=cancel outlined 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ dfDxFa"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;

exports[`button in auth page correct styles: variant=submit 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ jsEiLx"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;

exports[`button in auth page correct styles: variant=submit outlined 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ geUmsL"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;

exports[`button in auth page correct styles: variant=success 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ dbzJHP"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;

exports[`button in auth page correct styles: variant=success outlined 1`] = `
<div>
  <div
    class="react-ripples"
    style="position: relative; display: inline-flex; overflow: hidden;"
  >
    <button
      class="sc-bczRLJ hJaVMZ"
    >
      Войти
    </button>
    <s
      style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
    />
  </div>
</div>
`;
